src/drivers/distance_sensor/cm8jl65/cm8jl65.cpp

```diff
@@ -41,166 +41,142 @@
-	// Virtual destructor
+	/** Virtual destructor */
 	virtual ~CM8JL65() override;
 
+	/**
+	 * Method : init()
+	 * This method initializes the general driver for a range finder sensor.
+	 */
 	virtual int  init() override;
+
 	virtual int  ioctl(device::file_t *filp, int cmd, unsigned long arg) override;
 
 	/**
-	* Diagnostics - print some basic information about the driver.
-	*/
-	void				print_info();
+	 * Diagnostics - print some basic information about the driver.
+	 */
+	void print_info();
 
 private:
 
-	char 				             _port[20];
-	uint8_t                   _rotation;
-	float				             _min_distance;
-	float				             _max_distance;
-	int         	             _conversion_interval;
-	ringbuffer::RingBuffer	  *_reports;
-	int				               _fd;
-	uint8_t			             _linebuf[25];
-	uint8_t                   _cycle_counter;
+	/**
+	 * Reads data from serial UART and places it into a buffer
+	 */
+	int collect();
 
-	enum CM8JL65_PARSE_STATE	 _parse_state;
-	unsigned char             _frame_data[PARSER_BUF_LENGTH];
-	uint16_t                  _crc16;
-	int                       _distance_mm;
+	/**
+	 * Perform a reading cycle; collect from the previous measurement
+	 * and start a new one.
+	 */
+	void Run() override;
 
-	int				               _class_instance;
-	int				               _orb_class_instance;
+	/**
+	 * Initialise the automatic measurement state machine and start it.
+	 * @note This function is called at open and error time.  It might make sense
+	 *       to make it more aggressive about resetting the bus in case of errors.
+	 */
+	void start();
 
-	orb_advert_t			         _distance_sensor_topic;
+	/**
+	 * Stops the automatic measurement state machine.
+	 */
+	void stop();
 
-	perf_counter_t			       _sample_perf;
-	perf_counter_t			       _comms_errors;
+	char _port[20] {};
 
-	/**
-	* Initialise the automatic measurement state machine and start it.
-	*
-	* @note This function is called at open and error time.  It might make sense
-	*       to make it more aggressive about resetting the bus in case of errors.
-	*/
-	void				start();
+	unsigned char _frame_data[PARSER_BUF_LENGTH] {START_FRAME_DIGIT1, START_FRAME_DIGIT2, 0, 0};
 
-	/**
-	* Stop the automatic measurement state machine.
-	*/
-	void				stop();
+	int _class_instance{-1};
+	int _conversion_interval{CM8JL65_CONVERSION_INTERVAL};
+	int _fd{-1};
+	int _orb_class_instance{-1};
 
-	/**
-	 * Set the min and max distance thresholds.
-	*/
-	void				set_minimum_distance(float min);
-	void				set_maximum_distance(float max);
-	float			get_minimum_distance();
-	float			get_maximum_distance();
+	uint8_t _cycle_counter{0};
+	uint8_t _linebuf[25] {};
+	uint8_t _rotation{0};
 
-	/**
-	* Perform a reading cycle; collect from the previous measurement
-	* and start a new one.
-	*/
-	void				Run() override;
+	uint16_t _crc16{0};
 
-	int				collect();
+	float _max_distance{9.0f};
+	float _min_distance{0.10f};
 
-};
+	ringbuffer::RingBuffer *_reports{nullptr};
```

Comment:
Can this be gutted entirely (ringbuffer, ioctl, read, etc)?

Reply:
Unfortunately, not that easily... See my work [here](https://github.com/mcsauder/Firmware/commit/0b9f34bf811b3947c7a324fe2db976d13097f88c). However, if we can get this PR accepted and one to follow it from the continued work in the link I think the issue will become apparent. 

I can successfully deprecate IOCTL usage from I2C devices, but not CDev devices.  For CDev devices, [these calls](https://github.com/mcsauder/Firmware/blob/0b9f34bf811b3947c7a324fe2db976d13097f88c/src/drivers/distance_sensor/cm8jl65/cm8jl65.cpp#L636) are still required.

Reply:
Please disregard my comments above.  I discovered the issue preventing me from succeeding in deleting the ioctl calls was that the ioctl calls were where the start() method was being called from.  Palm->forehead. :}

I have pushed up a commit that is bench tested and accomplishes your requests.  I will post a/b comparisons and flight logs as soon as I can get to that work.